compiler-plugin/src/main/java/io/ballerina/stdlib/ftp/plugin/FTPServiceAnalysisTask.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.ftp.plugin;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ModuleSymbol;
+import io.ballerina.compiler.api.symbols.ServiceDeclarationSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.ballerina.stdlib.ftp.plugin.PluginUtils.validateModuleId;
+
+/**
+ * FTP service compilation analysis task.
+ */
+public class FTPServiceAnalysisTask implements AnalysisTask<SyntaxNodeAnalysisContext> {
+
+    private final FTPServiceValidator serviceValidator;
+
+    public FTPServiceAnalysisTask() {
+        this.serviceValidator = new FTPServiceValidator();
+    }
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+        List<Diagnostic> diagnostics = context.semanticModel().diagnostics();
+        for (Diagnostic diagnostic : diagnostics) {
+            if (diagnostic.diagnosticInfo().severity() == DiagnosticSeverity.ERROR) {
+                return;
+            }
+        }
+        if (!isFTPService(context)) {
+            return;
+        }
+        this.serviceValidator.validate(context);
+    }
+
+    private boolean isFTPService(SyntaxNodeAnalysisContext context) {
```

Comment:
Shall we rename these methods as `isFtpService`?
Please check other places as well.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-ftp/pull/558/commits/2ab5e744a42a7f4a71616b855ca16ef94533eb62